go/ngql/types.go

```diff
@@ -32,9 +37,29 @@ type getFieldFn func(v interface{}, fieldName string, ctx context.Context) types
 // returning one or more *noms* values whose presence is indicated by the provided arguments.
 type getSubvaluesFn func(v types.Value, args map[string]interface{}) (interface{}, error)
 
+// GraphQL requires all memberTypes in a Union to be Structs, so when a noms union contains a
+// scalar, we represent it in that context as a "boxed" value. E.g.
+// Boolean! =>
+// type BooleanValue {
+//   scalarValue: Boolean!
```

Comment:
Another option would be to do:

```
type BooleanValue {
  boolean: Boolean!
}

type NumberValue {
  number: Number!
}

type StringValue {
  string: String!
}
```

not sure if it would make simpler or more complicated to use?